### ISSUE TITLE
Debugging terminated. CORDBG_E_TARGET_INCONSISTENT, HRESULT=0x80131c36

### ISSUE THREAD

**Issue by mari-dot-tj** [dotnet/runtime#45547]
### Description
The debugger terminates returning:
_A fatal error has occurred and debugging needs to be terminated. For more details, please see the Microsoft Help and Support web site. HRESULT=0x80131c36. ErrorCode=0x0.The program ... has exited with code -1 (0xffffffff)._
I am having trouble replicating the problem because I'm running a larger application relying on other microservices. The error has accrued at multiple occasions and I found that someone else has the same issue: [https://github.com/dotnet/runtime/issues/2317#issuecomment-733682693](here). 

I was asked to post a new issue. Sorry that I cannot provide more information.

### Configuration
Running on macOS Catalina 10.15.7 (MacBook Pro 16inch late 2019) and Visual Studio for Mac (Professional) 8.8.3. Running .net Core 3.1 with SDK 3.1.404.

**Comment by tommcdon** [comment]
@hoyosjs

**Comment by nosami** [comment]
I hit this error frequently and means that I can't run my program under the debugger.

The process that I am running crashes after a seemingly random amount of time. I can't pinpoint why this happens or if it's specific code that is being debugged when it crashes unfortunately.

I'm on macOS Big Sur (11.0.1) VSMac 8.8.3, SDK 3.1.404

**Comment by rodrmoya** [comment]
Same here, happens for me on Mac (Catalina) with VSMac 8.8 and 8.9 and VSCode latest, and on Linux with VSCode, makes it impossible to use the debugger, as it crashes at random points, and the DEBUG_PROTOCOL output shows nothing, apart from the error message.

**Comment by hoyosjs** [comment]
@rodrmoya @nosami Is there any particular workload that you notice can repro this? Is there anything I can do to repro?

**Comment by nosami** [comment]
@hoyosjs Running the CloudEnvironmentsService.sln on this [repo](https://devdiv.visualstudio.com/OnlineServices/_git/vsclk-core) triggers a crash pretty quickly, but unfortunately, the steps required to be able to run this repo are fairly involved.

I guess I could add some logging code somewhere in [here](https://github.com/dotnet/coreclr/blob/v3.1.4/src/debug/daccess/dacdbiimpl.cpp#L3336-L3377) if I can figure out how to build and deploy the runtime

**Comment by hoyosjs** [comment]
There's a few places that can return such error (~40). I'll try to take a look, but right now there's quite a queue of things to take a look at. @tommcdon is helping me trying to repro this.

**Comment by nosami** [comment]
After downgrading to SDK 3.1.200, the issue goes away. I can test with a few other SDKs to see if I can help narrow down

**Comment by rodrmoya** [comment]
With 3.1.200 it was even worst for me, that's why I bumped locally to 3.1.40x, where the issue happens, but less often. Still, bad enough as it crashes in the middle of a debugging session at completely random time.

**Comment by tommcdon** [comment]
@rodrmoya @nosami It's been a while since we have update this issue.  Is this issue still reproducing?

**Comment by tommcdon** [comment]
This is an old issue is non-actionable, and so we are losing it until we have a project that is capable of reproducing the problem.

**Comment by rodrmoya** [comment]
we can easily repro on both Mac and Linux with a specific solution, which we can't share though, as it's private code. It got quite better a couple months ago, as we fixed a couple big men leaks on our code, but it's back, but now, after that improvement, it seems to be clear it might be in our code. Still the debugger shouldn't crash though, I guess, but at least it seems we can do something on our side, as we did a couple months ago.
